File: src/main/java/com/ultreon/ultranlang/symbol/BuiltinTypeSymbol.java
package com.ultreon.ultranlang.symbol;

public class BuiltinTypeSymbol extends Symbol {
    public static final String INTEGER = "INTEGER";
    public static final String REAL = "REAL";
    public static final String STRING = "STRING";

    public BuiltinTypeSymbol(String name) {
        super(name);
    }

    @Override
    public String toString() {
        return getName();
    }
}

File: src/main/kotlin/com/ultreon/ultranlang/symbol/BuiltinTypeSymbol.kt
package com.ultreon.ultranlang.symbol

class BuiltinTypeSymbol(name: String) : Symbol(name) {
    override fun toString(): String {
        return name
    }

    fun representation(): String {
        return "<${this::class.qualifiedName}(name=$name)>"
    }

    companion object {
        const val INTEGER: String = "INTEGER"
        const val REAL: String = "REAL"
        const val STRING: String = "STRING"
    }
}